## Cryptocurrency

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

#### Load Data

In [2]:
crypto_data = pd.read_csv('./Resources/crypto_data.csv')

In [3]:
crypto_data

Unnamed: 0        CoinName Algorithm  IsTrading ProofType  \
0            42         42 Coin    Scrypt       True   PoW/PoS   
1           365         365Coin       X11       True   PoW/PoS   
2           404         404Coin    Scrypt       True   PoW/PoS   
3           611       SixEleven   SHA-256       True       PoW   
4           808             808   SHA-256       True   PoW/PoS   
...         ...             ...       ...        ...       ...   
1247        XBC     BitcoinPlus    Scrypt       True       PoS   
1248       DVTC      DivotyCoin    Scrypt      False   PoW/PoS   
1249       GIOT     Giotto Coin    Scrypt      False   PoW/PoS   
1250       OPSC  OpenSourceCoin   SHA-256      False   PoW/PoS   
1251       PUNK       SteamPunk       PoS      False       PoS   

      TotalCoinsMined TotalCoinSupply  
0        4.199995e+01              42  
1                 NaN      2300000000  
2        1.055185e+09       532000000  
3                 NaN          611000  
4        0.000000e+00               0  
...               ...             ...  
1247     1.283270e+05         1000000  
1248     2.149121e+07       100000000  
1249              NaN       233100000  
1250              NaN        21000000  
1251              NaN        40000000  

[1252 rows x 7 columns]

#### Data Preparation

Removing cryptocurrency that is not currently being traded.

In [4]:
crypto_1 = crypto_data.loc[(crypto_data["IsTrading"]==True)]
crypto_1

Unnamed: 0     CoinName    Algorithm  IsTrading ProofType  \
0            42      42 Coin       Scrypt       True   PoW/PoS   
1           365      365Coin          X11       True   PoW/PoS   
2           404      404Coin       Scrypt       True   PoW/PoS   
3           611    SixEleven      SHA-256       True       PoW   
4           808          808      SHA-256       True   PoW/PoS   
...         ...          ...          ...        ...       ...   
1243       SERO   Super Zero       Ethash       True       PoW   
1244        UOS          UOS      SHA-256       True      DPoI   
1245        BDX       Beldex  CryptoNight       True       PoW   
1246        ZEN      Horizen     Equihash       True       PoW   
1247        XBC  BitcoinPlus       Scrypt       True       PoS   

      TotalCoinsMined TotalCoinSupply  
0        4.199995e+01              42  
1                 NaN      2300000000  
2        1.055185e+09       532000000  
3                 NaN          611000  
4        0.000000e+00               0  
...               ...             ...  
1243              NaN      1000000000  
1244              NaN      1000000000  
1245     9.802226e+08      1400222610  
1246     7.296538e+06        21000000  
1247     1.283270e+05         1000000  

[1144 rows x 7 columns]

In [5]:
crypto_2 = crypto_1.drop(columns=['IsTrading'])
crypto_2

Unnamed: 0     CoinName    Algorithm ProofType  TotalCoinsMined  \
0            42      42 Coin       Scrypt   PoW/PoS     4.199995e+01   
1           365      365Coin          X11   PoW/PoS              NaN   
2           404      404Coin       Scrypt   PoW/PoS     1.055185e+09   
3           611    SixEleven      SHA-256       PoW              NaN   
4           808          808      SHA-256   PoW/PoS     0.000000e+00   
...         ...          ...          ...       ...              ...   
1243       SERO   Super Zero       Ethash       PoW              NaN   
1244        UOS          UOS      SHA-256      DPoI              NaN   
1245        BDX       Beldex  CryptoNight       PoW     9.802226e+08   
1246        ZEN      Horizen     Equihash       PoW     7.296538e+06   
1247        XBC  BitcoinPlus       Scrypt       PoS     1.283270e+05   

     TotalCoinSupply  
0                 42  
1         2300000000  
2          532000000  
3             611000  
4                  0  
...              ...  
1243      1000000000  
1244      1000000000  
1245      1400222610  
1246        21000000  
1247         1000000  

[1144 rows x 6 columns]

Droping rows with at least one missing element.

In [6]:
crypto_3 = crypto_2.dropna()
crypto_3

Unnamed: 0     CoinName    Algorithm ProofType  TotalCoinsMined  \
0            42      42 Coin       Scrypt   PoW/PoS     4.199995e+01   
2           404      404Coin       Scrypt   PoW/PoS     1.055185e+09   
4           808          808      SHA-256   PoW/PoS     0.000000e+00   
5          1337    EliteCoin          X13   PoW/PoS     2.927942e+10   
7           BTC      Bitcoin      SHA-256       PoW     1.792718e+07   
...         ...          ...          ...       ...              ...   
1238       ZEPH       ZEPHYR      SHA-256      DPoS     2.000000e+09   
1242        GAP      Gapcoin       Scrypt   PoW/PoS     1.493105e+07   
1245        BDX       Beldex  CryptoNight       PoW     9.802226e+08   
1246        ZEN      Horizen     Equihash       PoW     7.296538e+06   
1247        XBC  BitcoinPlus       Scrypt       PoS     1.283270e+05   

     TotalCoinSupply  
0                 42  
2          532000000  
4                  0  
5       314159265359  
7           21000000  
...              ...  
1238      2000000000  
1242       250000000  
1245      1400222610  
1246        21000000  
1247         1000000  

[685 rows x 6 columns]

Removing Cryptocurrency that has not been mined.

In [7]:
crypto_4 = crypto_3.loc[(crypto_3["TotalCoinsMined"]>0)]
crypto_4

Unnamed: 0     CoinName    Algorithm ProofType  TotalCoinsMined  \
0            42      42 Coin       Scrypt   PoW/PoS     4.199995e+01   
2           404      404Coin       Scrypt   PoW/PoS     1.055185e+09   
5          1337    EliteCoin          X13   PoW/PoS     2.927942e+10   
7           BTC      Bitcoin      SHA-256       PoW     1.792718e+07   
8           ETH     Ethereum       Ethash       PoW     1.076842e+08   
...         ...          ...          ...       ...              ...   
1238       ZEPH       ZEPHYR      SHA-256      DPoS     2.000000e+09   
1242        GAP      Gapcoin       Scrypt   PoW/PoS     1.493105e+07   
1245        BDX       Beldex  CryptoNight       PoW     9.802226e+08   
1246        ZEN      Horizen     Equihash       PoW     7.296538e+06   
1247        XBC  BitcoinPlus       Scrypt       PoS     1.283270e+05   

     TotalCoinSupply  
0                 42  
2          532000000  
5       314159265359  
7           21000000  
8                  0  
...              ...  
1238      2000000000  
1242       250000000  
1245      1400222610  
1246        21000000  
1247         1000000  

[532 rows x 6 columns]

Deleting CoinName as the data must be numeric to be comprehensible to a machine learning algorithm and it does not contribute to analysis.

In [8]:
crypto_5 = crypto_4.drop(columns=['CoinName'])
crypto_5

Unnamed: 0    Algorithm ProofType  TotalCoinsMined TotalCoinSupply
0            42       Scrypt   PoW/PoS     4.199995e+01              42
2           404       Scrypt   PoW/PoS     1.055185e+09       532000000
5          1337          X13   PoW/PoS     2.927942e+10    314159265359
7           BTC      SHA-256       PoW     1.792718e+07        21000000
8           ETH       Ethash       PoW     1.076842e+08               0
...         ...          ...       ...              ...             ...
1238       ZEPH      SHA-256      DPoS     2.000000e+09      2000000000
1242        GAP       Scrypt   PoW/PoS     1.493105e+07       250000000
1245        BDX  CryptoNight       PoW     9.802226e+08      1400222610
1246        ZEN     Equihash       PoW     7.296538e+06        21000000
1247        XBC       Scrypt       PoS     1.283270e+05         1000000

[532 rows x 5 columns]

Using get_dummies() to convert features with text to numeric values. We have 5 columns and 532 rows.

In [11]:
crypto_6 = pd.get_dummies(crypto_5, columns=["Algorithm", "ProofType"])
crypto_6

Unnamed: 0  TotalCoinsMined TotalCoinSupply  \
0            42     4.199995e+01              42   
2           404     1.055185e+09       532000000   
5          1337     2.927942e+10    314159265359   
7           BTC     1.792718e+07        21000000   
8           ETH     1.076842e+08               0   
...         ...              ...             ...   
1238       ZEPH     2.000000e+09      2000000000   
1242        GAP     1.493105e+07       250000000   
1245        BDX     9.802226e+08      1400222610   
1246        ZEN     7.296538e+06        21000000   
1247        XBC     1.283270e+05         1000000   

      Algorithm_1GB AES Pattern Search  Algorithm_536  Algorithm_Argon2d  \
0                                    0              0                  0   
2                                    0              0                  0   
5                                    0              0                  0   
7                                    0              0                  0   
8                                    0              0                  0   
...                                ...            ...                ...   
1238                                 0              0                  0   
1242                                 0              0                  0   
1245                                 0              0                  0   
1246                                 0              0                  0   
1247                                 0              0                  0   

      Algorithm_BLAKE256  Algorithm_Blake  Algorithm_Blake2S  \
0                      0                0                  0   
2                      0                0                  0   
5                      0                0                  0   
7                      0                0                  0   
8                      0                0                  0   
...                  ...              ...                ...   
1238                   0                0                  0   
1242                   0                0                  0   
1245                   0                0                  0   
1246                   0                0                  0   
1247                   0                0                  0   

      Algorithm_Blake2b  ...  ProofType_PoW/PoS  ProofType_PoW/PoS   \
0                     0  ...                  1                   0   
2                     0  ...                  1                   0   
5                     0  ...                  1                   0   
7                     0  ...                  0                   0   
8                     0  ...                  0                   0   
...                 ...  ...                ...                 ...   
1238                  0  ...                  0                   0   
1242                  0  ...                  1                   0   
1245                  0  ...                  0                   0   
1246                  0  ...                  0                   0   
1247                  0  ...                  0                   0   

      ProofType_PoW/PoW  ProofType_PoW/nPoS  ProofType_Pos  \
0                     0                   0              0   
2                     0                   0              0   
5                     0                   0              0   
7                     0                   0              0   
8                     0                   0              0   
...                 ...                 ...            ...   
1238                  0                   0              0   
1242                  0                   0              0   
1245                  0                   0              0   
1246                  0                   0              0   
1247                  0                   0              0   

      ProofType_Proof of Authority  ProofType_Proof of Trust  ProofType_TPoS  \
0                   

Wow! The number of columns increased by 94!